src/kixi/stats/distribution.cljc

```diff
@@ -135,34 +135,50 @@
       :else (rand-binomial-btrd n p rng))))
 
 (defn ^:no-doc rand-gamma
-  [k rng]
-  (let [k' (cond-> k (< 1) inc)
-        a1 (- k' (/ 1 3))
-        a2 (/ 1 (sqrt (* 9 a1)))
+  "Returns a random variate generated from a Gamma distribution with shape
+  parameter `alpha`, internally using `rng` to generate random normal and
+  uniform variates.
+
+  The variate is generated using Marsaglia's transformation-rejection method
+  described in [\"A simple method for generating Gamma
+  variables\"](https://dl.acm.org/doi/10.1145/358407.358414), page 369.
+
+  ### References
+
+  - [Wikipedia section on random variate generation](https://en.wikipedia.org/wiki/Gamma_distribution#Random_variate_generation)"
+  [alpha rng]
+  (let [;; First part of the correction for $alpha < 1$, described on p371 of
+        ;; the paper.
+        alpha' (if (< alpha 1) (inc alpha) alpha)
```

Comment:
this correction was being applied to ALL values of alpha, since `(< 1)` was always true.